src/api/utils/openapi.decorator.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { applyDecorators, Header, HttpCode } from '@nestjs/common';
+import {
+  ApiBadRequestResponse,
+  ApiConflictResponse,
+  ApiCreatedResponse,
+  ApiInternalServerErrorResponse,
+  ApiNoContentResponse,
+  ApiNotFoundResponse,
+  ApiOkResponse,
+  ApiProduces,
+  ApiUnauthorizedResponse,
+} from '@nestjs/swagger';
+
+import { BaseDto } from '../../utils/base.dto.';
+import {
+  badRequestDescription,
+  conflictDescription,
+  createdDescription,
+  internalServerErrorDescription,
+  noContentDescription,
+  notFoundDescription,
+  okDescription,
+  unauthorizedDescription,
+} from './descriptions';
+
+export type HttpStatusCodes =
+  | 200
+  | 201
+  | 204
+  | 400
+  | 401
+  | 403
+  | 404
+  | 409
+  | 500;
+
+export interface HttpStatusCodeWithExtraInformation {
+  code: HttpStatusCodes;
+  description?: string;
+  isArray?: boolean;
+  dto?: BaseDto;
+  mimeType?: string;
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention,func-style
+export const OpenApi = (
```

Comment:
Could you write docs for this decorator?
It's quite "magical" so it is important to fully document what it does. Otherwise, we make it unnecessarily hard for newcomers to understand the codebase.